erts/emulator/beam/erl_map.c

```diff
@@ -125,15 +125,18 @@ BIF_RETTYPE map_size_1(BIF_ALIST_1) {
 	flatmap_t *mp = (flatmap_t*)flatmap_val(BIF_ARG_1);
 	BIF_RET(make_small(flatmap_get_size(mp)));
     } else if (is_hashmap(BIF_ARG_1)) {
-	Eterm *head, *hp, res;
-	Uint size, hsz=0;
+	Eterm *head;
+	Uint size;
 
 	head = hashmap_val(BIF_ARG_1);
 	size = head[1];
-	(void) erts_bld_uint(NULL, &hsz, size);
-	hp = HAlloc(BIF_P, hsz);
-	res = erts_bld_uint(&hp, NULL, size);
-	BIF_RET(res);
+
+        /*
+         * In practice, the size of a map will always fit in
+         * a small.
+         */
+        ASSERT(IS_USMALL(0, size));
```

Comment:
Use ERTS_ASSERT?

Reply:
I have reworded the comment and explained the purpose of the ASSERT (which is to catch a problem if we would ever reduce the number of useful bits in small).